Debian Packaging: Only recommend apache2

ChristianTackeGSI opened the pull request:
cvmfs-server currently strictly depends on apache2.
On a publisher node, one does not need apache2.
So turn the "Depends: apache2" into a "Recommends: apache2".

Recommends means "You really should install this. If you don't, expect missing functionality". Which fits very well: If one does not install apache2, one can't run a stratum-0.

ChristianTackeGSI committed:
Debian Packaging: Only recommend apache2

cvmfs-server currently strictly depends on apache2.
On a publisher node, one does not need apache2.
So turn the "Depends: apache2" into a
"Recommends: apache2".

Recommends means "You really should install this. If you
don't, expect missing functionality". Which fits very well:
If one does not install apache2, one can't run a stratum-0.

jblomer commented:
@cernvm-bot cloudtest

jblomer reviewed (changes requested):
Many thanks!  Looks in principle good to me.  Could you also add `apache2` as an explicit package to test/cloud_testing/platforms/ubuntu_setup.sh

That should fix the integration tests

ChristianTackeGSI committed:
Explicitly install apache2 in Ubuntu tests

As apache2 now is a Recommends, it looks like it has
to be installed explicitly in the testsuite.

ChristianTackeGSI commented:
done.

I have not tested this! So I hope your CI will do that now.

jblomer commented:
@cernvm-bot cloudtest

jblomer commented:
Thanks!  It won't make it in the 2.7.4 patch release that we are cutting just now but it will be part of the 2.8 release.

ChristianTackeGSI commented:
> Thanks! It won't make it in the 2.7.4 patch release that we are cutting just now but it will be part of the 2.8 release.

Thanks! 2.8 is fine. It will just simplify our setup a bit.
